qemu/tests/bridge_vlan.py

```diff
@@ -0,0 +1,239 @@
+            err_msg = "Could not log into guest %s" % vm.name
+            test.error(err_msg)
+        sessions.append(session)
+        logging.info("Logged in %s successful" % vm.name)
+
+        if os_type == "linux":
+            ifname.append(utils_net.get_linux_ifname(session,
+                                                     vm.get_mac_address()))
+
+            error_context.context("Load 8021q module in guest %s" % vm.name,
+                                  logging.info)
+            session.cmd_output_safe("modprobe 8021q")
+
+            error_context.context("Setup vlan environment in guest %s" % vm.name,
+                                  logging.info)
+            inter_ip = "%s.%s.%d" % (subnet, host_vlan_id, vm_index + 1)
+            set_ip_vlan("eth0", inter_ip, session=session)
+            vm_ip.append(inter_ip)
+            set_arp_ignore(session)
+            error_context.context("Test ping from guest '%s' to host with interface '%s'" %
+                                  (vm.name, ifname[vm_index]), logging.info)
+            try:
+                ping_vlan(vm, dest=host_vlan_ip, vlan_if='eth0', session=session)
+            except Exception:
+                logging.info("Guest should not ping success to host with interface '%s'" %
+                             ifname[vm_index])
+
+    # Ping succeed between guests with eth0
+    error_context.context("Test ping between guests with interface '%s'" %
+                          ifname[0], logging.info)
+    ping_vlan(vms[0], dest=vm_ip[1], vlan_if=ifname[0], session=sessions[0])
+
+    # set vlan tag for guest
+    for vm_index, vm in enumerate(vms):
+        session = sessions[vm_index]
+        error_context.context("Add vlan interface on guest '%s'" % vm.name)
+        session.cmd("ifconfig %s 0.0.0.0" % ifname[vm_index])
+        vlan_if = add_vlan(interface=ifname[vm_index], v_id=host_vlan_id, session=session)
+        vm_vlan_if.append(vlan_if)
+        set_mac_vlan(vlan_if, mac_str[vm_index], session=session)
+        vlan_ip = "%s.%s.%d" % (subnet, host_vlan_id, vm_index + 11)
+        set_ip_vlan(vlan_if, vlan_ip, session=session)
+        vm_vlan_ip.append(vlan_ip)
+
+        error_context.context("Test ping from interface '%s' on guest '%s' to host." %
+                              (vm_vlan_if[vm_index], vm.name), logging.info)
+        ping_vlan(vm, dest=host_vlan_ip, vlan_if=vm_vlan_if[vm_index], session=session)
+        netperf_vlan(client=vm.name, server="localhost")
+
+    error_context.context("Test ping and netperf bwtween guests with interface '%s'" %
+                          vm_vlan_if[0], logging.info)
+    ping_vlan(vms[0], dest=vm_vlan_ip[1], vlan_if=vm_vlan_if[0], session=sessions[0])
+    netperf_vlan(client=params["main_vm"], server='vm2')
+
+    exithandlers = "exithandlers__%s" % sub_type
+    sub_exit_timeout = int(params.get("sub_exit_timeout", 10))
+    start_time = time.time()
+    end_time = time.time() + float(sub_exit_timeout)
```

Comment:
end_time = start_time + float(sub_exit_timeout)

Reply:
Looks more readable, updated it, thanks.